app/taco.ts

```diff
@@ -0,0 +1,97 @@
+import {conditions, decrypt, domains, encrypt, ThresholdMessageKit} from '@nucypher/taco';
+import { EIP4361AuthProvider, USER_ADDRESS_PARAM_DEFAULT } from '@nucypher/taco-auth';
+import {ethers} from "ethers";
+
+let authManager = {
+    provider: null,
+    authProvider: null,
+    currentAccount: null,
+
+    async initialize() {
+      if (!this.provider) {
+        this.provider = new ethers.providers.Web3Provider(window.ethereum);
+        await this.updateAccount();
+        this.setupAccountChangeListener();
+      }
+      return this.authProvider;
```

Comment:
Analogous in `taco-composedb` - https://github.com/nucypher/taco-composedb/blob/main/utils/index.ts#L19.

Reply:
Be careful because initialize is only called once, so if `window.ethereum` is not defined at that time, you will never have a `this.provider` that is set.

Reply:
@manumonti how did you run the app?

Reply:
i added a retry mechanism for `window.ethereum`

Reply:
> @manumonti how did you run the app? i haven't seen this error at all

With `pnpm dev`. Maybe is something related to the node version that is being used? Here are the full logs:

https://gist.github.com/manumonti/d00bdd177bda546927d25c2c281fc320